x-pack/plugins/cases/public/components/user_actions/comment/comment.tsx

```diff
@@ -0,0 +1,371 @@
+  className: classNames('userAction__comment', {
+    outlined,
+    isEdit,
+  }),
+  children: (
+    <UserActionMarkdown
+      ref={(element) => (commentRefs.current[comment.id] = element)}
+      id={comment.id}
+      content={comment.comment}
+      isEditable={isEdit}
+      onChangeEditable={handleManageMarkdownEditId}
+      onSaveContent={handleSaveComment.bind(null, {
+        id: comment.id,
+        version: comment.version,
+      })}
+    />
+  ),
+  timelineIcon: (
+    <UserActionAvatar username={comment.createdBy.username} fullName={comment.createdBy.fullName} />
+  ),
+  actions: (
+    <UserActionContentToolbar
+      id={comment.id}
+      commentMarkdown={comment.comment}
+      editLabel={i18n.EDIT_COMMENT}
+      quoteLabel={i18n.QUOTE}
+      isLoading={isLoading}
+      onEdit={handleManageMarkdownEditId.bind(null, comment.id)}
+      onQuote={handleManageQuote.bind(null, comment.comment)}
+      userCanCrud={userCanCrud}
+    />
+  ),
+});
+
+const getAlertAttachmentUserAction = ({
+  userAction,
+  comment,
+  alertData,
+  getRuleDetailsHref,
+  loadingAlertData,
+  onRuleDetailsClick,
+  onShowAlertDetails,
+}: {
+  userAction: UserActionResponse<CommentUserAction>;
+  comment: SnakeToCamelCase<CommentResponseAlertsType>;
+  alertData: Record<string, Ecs>;
+  getRuleDetailsHref?: RuleDetailsNavigation['href'];
+  onRuleDetailsClick?: RuleDetailsNavigation['onClick'];
+  loadingAlertData: boolean;
+  onShowAlertDetails: (alertId: string, index: string) => void;
+}): EuiCommentProps => {
+  // TODO: clean this up
+  const alertId = Array.isArray(comment.alertId)
+    ? comment.alertId.length > 0
+      ? comment.alertId[0]
+      : ''
+    : comment.alertId;
+
+  const alertIndex = Array.isArray(comment.index)
+    ? comment.index.length > 0
+      ? comment.index[0]
+      : ''
+    : comment.index;
+
+  // if (isEmpty(alertId)) {
+  //   return comments;
+  // }
+
+  const ruleId =
+    comment?.rule?.id ??
+    alertData[alertId]?.signal?.rule?.id?.[0] ??
+    get(alertData[alertId], ALERT_RULE_UUID)[0] ??
+    null;
+
+  const ruleName =
+    comment?.rule?.name ??
+    alertData[alertId]?.signal?.rule?.name?.[0] ??
+    get(alertData[alertId], ALERT_RULE_NAME)[0] ??
+    null;
+
+  return {
+    username: (
+      <UserActionUsernameWithAvatar
+        username={userAction.createdBy.username}
+        fullName={userAction.createdBy.fullName}
+      />
+    ),
+    className: 'comment-alert',
+    type: 'update',
+    event: (
+      <AlertCommentEvent
+        alertId={alertId}
+        getRuleDetailsHref={getRuleDetailsHref}
+        loadingAlertData={loadingAlertData}
+        onRuleDetailsClick={onRuleDetailsClick}
+        ruleId={ruleId}
+        ruleName={ruleName}
+        commentType={CommentType.alert}
+      />
+    ),
+    'data-test-subj': `${userAction.type}-${userAction.action}-action-${userAction.actionId}`,
```

Comment:
nitpick: Always put a static-string on your `data-test-subj`. It makes it easier to understand it when you are reading the code or seeing the rendered output.

Reply:
This can be rendered multiple times. Do you think is best to have multiple elements with the same `data-test-subj`?

Reply:
@cnasikas I don't mean to replace the variables with static strings. Simply to add a static suffix keeping the dynamic part too.

Reply:
e.g.

```
     'data-test-subj': `use-action-link-${userAction.type}-${userAction.action}-action-${userAction.actionId}`, 
``

Reply:
Oh, I see what you mean. Yes, ofc!